analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -0,0 +1,284 @@
+                                                  "callers_per_mutation.rds"))
+```
+
+## Upset graph
+
+Make the upset plot. 
+
+```{r}
+# Make the plot
+upset <- UpSetR::upset(UpSetR::fromList(mutation_id_list),
+  order.by = "freq",
+  text.scale = 2
+)
+
+# Save this plot
+png(file.path(consensus_plots_dir, "upset_plot.png"), width = 900, height = 900)
+upset
+dev.off()
+
+# Print this out
+upset
+```
+
+## VAF for combination sets of callers
+
+Graph mutations by combinations of callers and plot the VAF density.
+Get list of callers per mutation and the median vaf for each. 
+Graph the median VAF for each combination of callers.
+
+```{r}
+# Make this plot
+callers_per_mutation %>%
+  ggplot2::ggplot(ggplot2::aes(x = caller_combo, y = median_vaf)) +
+  ggplot2::geom_violin() +
+  ggplot2::theme_classic() +
+  ggupset::scale_x_mergelist(sep = "-") +
+  ggupset::axis_combmatrix(sep = "-") +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Median VAF Across Callers")
+```
+
+```{r}
+# Save this plot
+ggplot2::ggsave(file.path(consensus_plots_dir, "upset_median_vaf_plot.png"))
+```
+
+## Base Changes 
+
+Set up the summarized form of the base change information. 
+
+```{r}
+# Summarize by the number of times each base change shows up in each category.
+perc_change_df <- vaf_df %>%
+  dplyr::count(caller, change, name = "count") %>%
+  dplyr::add_count(caller, wt = count) %>%
+  dplyr::mutate(percent = count / n) %>%
+  dplyr::filter(!grepl("G>G|A>A|N>A|N>T", change))
+
+# Need to rearrange the order of the base change categories
+perc_change_df <- perc_change_df %>%
+  dplyr::mutate(
+    change = as.factor(change),
+    change = factor(change, c(grep("ins|del|long_change", levels(change), value = TRUE, invert = TRUE), "ins", "del", "long_change"))
+  )
+```
+
+Make a barplot illustrating the percent of the mutations for each caller that 
+that are each type of change. 
+
+```{r}
+perc_change_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = change, y = percent)) +
+  ggplot2::theme_classic() +
+  ggplot2::geom_bar(
+    position = "dodge", stat = "identity",
+    ggplot2::aes(fill = caller)
+  ) +
+  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Percent of callers' mutations") +
+  colorblindr::scale_fill_OkabeIto()
+```
+
+```{r}
+# Save this plot
+ggplot2::ggsave(file.path(consensus_plots_dir, "perc_base_change_plot.png"))
+```
+
+## VAF distributions
+
+```{r}
+vaf_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = caller, y = vaf, color = caller)) +
+  ggplot2::geom_violin() +
+  ggplot2::theme_classic() +
+  colorblindr::scale_color_OkabeIto()
+```
+
+```{r}
+# Save this plot
+ggplot2::ggsave(file.path(consensus_plots_dir, "vaf_violin_plot.png"))
+```
+
+## TMB distributions
+
+```{r}
+tmb_df %>%
+  ggplot2::ggplot(ggplot2::aes(x = caller, y = tmb, color = short_histology)) +
+  ggplot2::geom_boxplot() +
+  ggplot2::theme_classic()
+```
```

Comment:
If you decide to change the figure dimensions above, you may want to change them here too (for the purpose of make it a bit easier to read on the html output).

Reply:
Which dimensions?